File: backend/Services/Parsers/Moviedle2Parser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class Moviedle2Parser : BasicScoreResultParser
    {
        private ILogger<Moviedle2Parser> _logger;

        public Moviedle2Parser(ILogger<Moviedle2Parser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override bool Default => false;
        public override string GameName => "Moviedle 2";
        public override string? HelpText => "Guess a movie based on its similarities with your prior guesses.";
        protected override Regex Parser => new Regex($"Moviedle #\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => "Play here: ";
        public override string Url => "https://moviedle.xyz";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace(ExtraContent, string.Empty).Replace(Url, string.Empty).Trim();
        }
    }
}

File: lambdas/src/parsers/moviedle2-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class Moviedle2Parser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Moviedle 2';
  readonly helpText = 'Guess a movie based on its similarities with your prior guesses.';
  protected readonly extraContent = 'Play here: ';
  readonly url = 'https://moviedle.xyz';
  
  // Matches "Moviedle #471 X/8" or "Moviedle #1186 1/8"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `Moviedle #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace(this.extraContent ?? '', '')
      .replace(this.url ?? '', '')
      .trim();
  }
}
